src/workflow_provider/default_vm_workflow_provider.cpp

```diff
@@ -50,24 +50,36 @@ auto workflows_map_for(const std::string& archive_file_path)
 {
     std::map<std::string, std::string> workflows_map;
     std::ifstream zip_stream{archive_file_path, std::ios::binary};
-    Poco::Zip::ZipArchive zip_archive{zip_stream};
-
-    for (auto it = zip_archive.headerBegin(); it != zip_archive.headerEnd(); ++it)
+    try
     {
-        if (it->second.isFile())
-        {
-            auto file_name = it->second.getFileName();
-            QFileInfo file_info{QString::fromStdString(file_name)};
+        auto zip_archive = MP_POCOZIPUTILS.zip_archive_for(zip_stream);
 
-            if (file_info.path().contains(workflow_dir_version) && file_info.suffix() == "yaml")
+        for (auto it = zip_archive.headerBegin(); it != zip_archive.headerEnd(); ++it)
+        {
+            if (it->second.isFile())
             {
-                Poco::Zip::ZipInputStream zip_input_stream{zip_stream, it->second};
-                std::ostringstream out(std::ios::binary);
-                Poco::StreamCopier::copyStream(zip_input_stream, out);
-                workflows_map[file_info.baseName().toStdString()] = out.str();
+                auto file_name = it->second.getFileName();
+                QFileInfo file_info{QString::fromStdString(file_name)};
+
+                if (file_info.path().contains(workflow_dir_version) &&
+                    (file_info.suffix() == "yaml" || file_info.suffix() == "yml"))
```

Comment:
`badv1yaml.yaml` would be considered OK here. Should be strict about `^v1/([a-z0-9-]+).(yaml|yml)$` or so.

Reply:
I still have yet to implement the validation of the file name as you suggested previously.  This is **only** to allow the `yaml` and `yml` suffixes.

Reply:
But yeah, I see what you mean about validating the whole path a little better :smiley:

Reply:
Ok, looking a this some more, sure, `badv1yaml.yaml` would indeed be considered OK here as long as the **path** (not just the file name) has `v1` in it based on [`QFileInfo::path()`](https://doc.qt.io/qt-5/qfileinfo.html#path).  I could a bit better here to make sure the parent directory of the file is indeed `v1`, so instead of `QFileInfo::path().contains()`, I should use `QFileInfo::dir().dirName() == workflow_dir_version`.

Reply:
This is done.